### PR TITLE
Fixed the implementation of learning from unlabeled data.

### PR DESCRIPTION
**Pull request by zacker150**
You seem to have mis-read the paper. According to the paper, you are supposed to have each classifier independently label previously unlabeled data and add them to the labled set. 

For reference, please see excerpt from the paper:

this

### CODE REVIEW

File: sklearn_cotraining/classifiers.py [hunk 1]
```diff
@@ -128,11 +127,7 @@ def fit(self, X1, X2, y):
 			L.extend([U_[x] for x in p])
 			L.extend([U_[x] for x in n])
 
-			#TODO: optimize these removals from U_
-			#this is currently (2p + 2n)O(n)
-			#and I think it can be reduced to O(n) rather easily
-			for i in p: U_.pop(i)
-			for i in n: U_.pop(i)
+			U_ = [elem for elem in U_ if (elem in p or elem in n)]
```
Comment (jjrob13) [inline sklearn_cotraining/classifiers.py]: Should we be populating U_ with new samples (i.e. samples that are not in p or n - which were just added into L)?

File: sklearn_cotraining/classifiers.py [hunk 2]
```diff
@@ -128,11 +127,7 @@ def fit(self, X1, X2, y):
...
 
 			#add new elements to U_
 			add_counter = 0 #number we have added from U to U_
```
Comment (jjrob13) [inline sklearn_cotraining/classifiers.py:133]: Do we still need this block here to refill U_?

**Review by jjrob13 — COMMENTED** [review]
Looks like you are correct!  Thanks for the PR, it's been a while since I've looked at this.  I was hoping to clarify a few things, but looks like a necessary change, overall.  Thanks again!

File: sklearn_cotraining/classifiers.py (hunk 1 above)
Comment (zacker150) [reply sklearn_cotraining/classifiers.py]: You are right. That line was supposed to remove all elements of p and n from U_. That has been fixed.

File: sklearn_cotraining/classifiers.py (hunk 2 above)
Comment (zacker150) [reply sklearn_cotraining/classifiers.py:133]: We are still using the old code to refill U_.

**Review by jjrob13 — APPROVED** [review]
Looks good! Thanks for the change!

**Pull request merged**